### PR TITLE
added sphinx style docstrings to all methods outside classes 

### LINKED ISSUE #14: Document league.py
Add docstrings in [this format](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html).

### PR DESCRIPTION
**Pull request by isobarbaric**
Fixes: #14 

added sphinx style docstrings to all methods outside classes (except for the League class) and wherever I wasn't sure about what a parameter was for, I put ``(not sure)`` next to those parameters to bring those to the attention of others who may know what those parameters are

other than that, my apologies if I did something wrong, I hope my edits will be useful :)

P.S. this is my first PR :D

### CODE REVIEW

File: pyleague/league.py [hunk 1]
```diff
@@ -3,12 +3,23 @@
...
-class League:
+class League:    
+    """This is a class that conceptually emulates a real-world football league. It contains the attributes and methods that would be necessary in a real-world league. 
+
+    :param name: name of the league 
+    :type name: str
+    :param teams: list of Team objects
+    :type teams: list
+    ...
+    :raises ValueError: if the team objects are not all unique in terms of their .identifier attribute, then a ValueError will be raised
```
Comment (nicolafan) [inline pyleague/league.py:17]:
```suggestion
    :raises ValueError: if the team objects are not all unique in terms of their .identifier attribute
```
No need to repeat this :)

File: pyleague/league.py [hunk 2]
```diff
@@ -64,12 +77,30 @@ def _generate_fixtures(self):
...
+        :type identifier: str
+        ...
+        :return: Participant object's whose Team object (in the self.team attribute) identifier matches that of the identifier parameter
+        :rtype: Participant
+        """
         return [x for x in self.participants if x.team.identifier == identifier][0]
 
     def generate_matchday(self):
+        """
+        """
```
Comment (nicolafan) [inline pyleague/league.py:100]:
```suggestion
        """Generates results for entire matchday
        """
```

**Comment by nicolafan** [comment]
Thank you so much for this PR! Your import changes were correct, I didn't know the imports I had were incorrect because they were working on my system (probably because I was using PyCharm with some kind of config.).
Your edits will be super-useful! I hope you didn't have problems understanding methods etc.
Thank you for your first PR :) :100: :1st_place_medal:

**Comment by isobarbaric** [comment]
Thank you so much for accepting my PR! Yeah, I changed the imports back to what you had originally since I thought what you had was what you wanted to go wth :). Nah, not all, I understood most of the stuff :) 

This project is super cool!! Hope to continue contributing to this project in the future :D

**Comment by nicolafan** [comment]
Sure! A New PR is always welcome! :)

**Pull request merged**